pkg/infra/tracing/tracing.go

```diff
@@ -109,6 +112,26 @@ func TraceIDFromContext(ctx context.Context, requireSampled bool) string {
 	return spanCtx.TraceID().String()
 }
 
+// RecordStatusError helper function to set the status of the current span to error and
+// record the error as an exception in the span.
+func RecordStatusError(ctx context.Context, err error, statusMessage ...string) error {
```

Comment:
we use a function internally (in HG) like this - I quite like the semantics as it's an almost-drop-in replacement for `fmt.Errorf`:

```go
// Errorf wraps fmt.Errorf and also logs the error in the trace span.
func Errorf(span trace.Span, format string, args ...interface{}) error {
	err := fmt.Errorf(format, args...)
	span.RecordError(err)
	span.SetStatus(codes.Error, err.Error())
	return err
}
```

Reply:
Nice thanks 👍  I have never understood why you normally end up putting the error as an exception event and at the same time marking the status with the same information (error string). That's why I tried to put msg as status description without normally the detailed error 🤷